viewer/src/datamodels/cad/sector/CachedRepository.ts

```diff
@@ -164,11 +164,13 @@ export class CachedRepository implements Repository {
     if (cached !== undefined) {
       return cached;
     }
-    // TODO 2021-05-05 larsmoa: Move retry to getBinaryFile()
-    const buffer = await this._modelSectorProvider.getBinaryFile(modelBlobUrl, filename);
-    const parsedCtm = await this._modelDataParser.parseCTM(new Uint8Array(buffer));
-    this._ctmFileCache.set(cacheKey, parsedCtm);
-    return parsedCtm;
+    const ctmPromise = (async () => {
+      const buffer = await this._modelSectorProvider.getBinaryFile(modelBlobUrl, filename);
+      const parsedCtm = await this._modelDataParser.parseCTM(new Uint8Array(buffer));
+      return parsedCtm;
+    })();
```

Comment:
```suggestion
    const ctmPromise = this._modelSectorProvider.getBinaryFile(modelBlobUrl, filename)
      .then((buffer) => this._modelDataParser.parseCTM(new Uint8Array(buffer)))
    ;
```

though I think a good old promise chain looks better than self invoking function